x-pack/legacy/plugins/graph/public/render_app.ts

```diff
@@ -0,0 +1,145 @@
+import {
+  AppMountContext,
+  ChromeStart,
+  SavedObjectsClientContract,
+  ToastsStart,
+  UiSettingsClientContract,
+} from 'kibana/public';
+// @ts-ignore
+import { initGraphApp } from './app';
+import { Plugin as DataPlugin } from '../../../../../src/plugins/data/public';
+
+/**
+ * These are dependencies of the Graph app besides the base dependencies
+ * provided by the application service. Some of those still rely on non-shimmed
+ * plugins in LP-world, but if they are migrated only the import path in the plugin
+ * itself changes
+ */
+export interface GraphDependencies extends LegacyAngularInjectedDependencies {
+  element: HTMLElement;
+  appBasePath: string;
+  capabilities: Record<string, boolean | Record<string, boolean>>;
+  coreStart: AppMountContext['core'];
+  chrome: ChromeStart;
+  config: UiSettingsClientContract;
+  toastNotifications: ToastsStart;
+  indexPatterns: DataStart['indexPatterns']['indexPatterns'];
+  npData: ReturnType<DataPlugin['start']>;
+  savedObjectsClient: SavedObjectsClientContract;
+  xpackInfo: { get(path: string): unknown };
+  addBasePath: (url: string) => string;
+  getBasePath: () => string;
+  Storage: any;
+  canEditDrillDownUrls: boolean;
+  graphSavePolicy: string;
+}
+
+/**
+ * Dependencies of the Graph app which rely on the global angular instance.
+ * These dependencies have to be migrated to their NP counterparts.
+ */
+export interface LegacyAngularInjectedDependencies {
+  /**
+   * angular $http service
+   */
+  $http: any;
+  /**
+   * Instance of SavedObjectRegistryProvider
+   */
+  savedObjectRegistry: any;
+  kbnBaseUrl: any;
+  /**
+   * Instance of SavedWorkspacesProvider
+   */
+  savedGraphWorkspaces: any;
+}
+
+export const renderApp = ({ appBasePath, element, ...deps }: GraphDependencies) => {
+  const graphAngularModule = createLocalAngularModule(deps.coreStart);
+  configureAppAngularModule(graphAngularModule);
+  initGraphApp(graphAngularModule, deps);
+  const $injector = mountGraphApp(appBasePath, element);
+  return () => $injector.get('$rootScope').$destroy();
+};
+
+const mainTemplate = (basePath: string) => `<div style="height: 100%">
+  <base href="${basePath}" />
+  <div ng-view style="height: 100%; display:flex; justify-content: center;"></div>
+</div>
+`;
+
+const moduleName = 'app/graph';
+
+const thirdPartyAngularDependencies = ['ngSanitize', 'ngRoute', 'react', 'ui.bootstrap', 'ui.ace'];
+
+function mountGraphApp(appBasePath: string, element: HTMLElement) {
+  const mountpoint = document.createElement('div');
+  mountpoint.setAttribute('style', 'height: 100%');
+  // eslint-disable-next-line
+  mountpoint.innerHTML = mainTemplate(appBasePath);
+  // bootstrap angular into detached element and attach it later to
+  // make angular-within-angular possible
+  const $injector = angular.bootstrap(mountpoint, [moduleName]);
+  element.appendChild(mountpoint);
+  return $injector;
+}
+
+function createLocalAngularModule(core: AppMountContext['core']) {
```

Comment:
Nit: since core is not used, it can be removed, also you could directly return the modules without `graphAngularModule`